### PR TITLE
[DOC]Groupby examples documentation

### LINKED ISSUE #2363: Documentation Request: using the new groupby expressions feature
I have been v interested in this feature since datatable came out - and now it's here! (actually since November).   While I do plan to experiment with it an actual list of illustrative usages would be quite handy to add somewhere to the docs.  The authors may know about nifty abilities/usages that I would not likely discover just ad-hoc.  Thanks!

### PR DESCRIPTION
**Pull request by samukweku**
This is a collection of examples for groupby - it aims to improve the documentation for datatable and show various uses of the `by` function.

Closes #2363

### CODE REVIEW

**Comment by samukweku** [comment]
Do I need to manually add my name to the list of contributors? If so, what does the number next to the names signify? number of approved PRs?

**Comment by st-pasha** [comment]
The list of contributors is generated semi-automatically via script `ci/gh.py`. Once you run the script for the correct milestone, it will report all contributors, which then needs to be pasted into the changelog file. The process cannot be fully automated because it would have to modify the files in the repository, which has lots of problems.

In general, we would run the script as part of the release process to ensure that the counts are accurate; but it can also be run before then to update the current contributor list.

**Comment by samukweku** [comment]
I read that part in the docs, and it was not too clear. I surmised then that the script does the update of the contributors. I suggest, if that is ok with you, that that section be made clearer, so contributors know that they don’t need to include their names in the contributor section, as that would be covered by the project maintainers via the script

File: docs/manual/groupby_examples.rst [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+                01-02   TMAX    71
+                01-02   TMIN    26"""
+
+    # Task : group by Day and find minimum Data_Value for TMIN 
+    # and maximum Data_Value for TMAX
+
+    df[:,
+       f.Day.extend({"TMAX" : ifelse(f.Element=="TMAX", f.Data_Value, None),
+                     "TMIN" : ifelse(f.Element=="TMIN", f.Data_Value, None)})
+      ][:, [max(f.TMAX), min(f.TMIN)], by('Day')]
```
Comment (st-pasha) [inline docs/manual/groupby_examples.rst:642]: You can combine these two expressions into a single one:
```
df[:, {"TMAX": max(ifelse(f.Element=="TMAX", f.Data_Value, None)),
       "TMIN": min(ifelse(f.Element=="TMIN", f.Data_Value, None))},
   by(f.Day)]
```

**Comment by samukweku** [comment]
Thanks for the feedback @st-pasha , I"ll make the edits and push

**Comment by st-pasha** [comment]
Thanks Samuel, looks great now.

**Pull request merged**